website/docs/plugin/framework/migrating/providers/index.mdx

```diff
@@ -0,0 +1,379 @@
+provider.
+
+For a complete example, clone the
+`terraform-provider-tls` repository and compare `provider.go` in
+[v3.4.0](https://github.com/hashicorp/terraform-provider-tls/blob/v3.4.0/internal/provider/provider.go)
+with [v4.0.1](https://github.com/hashicorp/terraform-provider-tls/blob/v4.0.1/internal/provider/provider.go).
+
+#### SDKv2
+
+The following example shows how to set up a provider schema, configuration, resources, and data sources using SDKv2.
+
+```go
+func New() (*schema.Provider, error) {
+    return &schema.Provider{
+        Schema: map[string]*schema.Schema{
+            "proxy": {
+            /* ... */
+            },
+        },
+        ConfigureContextFunc: configureProvider,
+        ResourcesMap: map[string]*schema.Resource{
+            "tls_private_key": resourcePrivateKey(),
+            /* ... */
+        },
+        DataSourcesMap: map[string]*schema.Resource{
+            "tls_public_key": dataSourcePublicKey(),
+            /* ... */
+        },
+    }, nil
+}
+```
+
+#### Framework
+
+The following shows the same section of provider code after the migration.
+
+```go
+var _ provider.Provider = (*provider)(nil)
+
+func New() provider.Provider {
+    return &provider{}
+}
+
+func (p *provider) GetResources(_ context.Context) (map[string]provider.ResourceType, diag.Diagnostics) {
+    return map[string]provider.ResourceType{
+        "tls_private_key":         &privateKeyResourceType{},
+        /* ... */
+    }, nil
+}
+
+func (p *provider) GetDataSources(_ context.Context) (map[string]provider.DataSourceType, diag.Diagnostics) {
+    return map[string]provider.DataSourceType{
+        "tls_public_key":  &publicKeyDataSourceType{},
+        /* ... */
+    }, nil
+}
+
+func (p *provider) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+    return tfsdk.Schema{
+        Attributes: map[string]tfsdk.Attribute{
+            "proxy": {
+            /* ... */
+            },
+        },
+    }, nil
+}
+
+func (p *provider) Configure(ctx context.Context, req provider.ConfigureRequest, res *provider.ConfigureResponse) {
+    /* ... */
+}
+```
+
+## Provider Schema
+
+A provider schema defines the attributes and behaviors of the provider itself. For example, a provider that connects to
+a third-party API may define attributes for the base URL or a required authentication token.
+
+### SDKv2
+
+In SDKv2, you implement a provider Schema by populating the `Schema` field on the `schema.Provider` struct. The `Schema`
+field contains a `map[string]*schema.Schema`. Each map entry represents the name of the attribute and pointer to a
+`schema.Schema` struct that defines that attribute's behavior.
+
+The following example defines the provider schema in the `Schema` field within the `schema.Provider` struct.
+
+```go
+func New() *schema.Provider {
+    return &schema.Provider{
+        Schema: map[string]*schema.Schema{},
+        /* ... */
```

Comment:
```suggestion
func New() *schema.Provider {
    return &schema.Provider{
        Schema: map[string]*schema.Schema{
            /* ... */
        },
```
Just to re-enforce that the stuff mentioned above goes inside the map definition.